docs/wave.rst

```diff
@@ -52,55 +53,94 @@ First of all, we will set up the initial condition for damping::
 
 DenseData is a devito data object used to store and manage spatially varying data.
 
-:samp:`damp_boundary()` function initialises the damp on each grid point. The initial data are stored in :obj:`self.damp.data`.
+:samp:`damp_boundary()` function initialises the damp on each grid point.
+The dampening values in data are stored in :obj:`self.damp.data`.
 
-Similarly, we will set up the initial condition and allocate the grid for m.
-Initial value of m on each grid point is stored as a numpy array in :obj:`m.data[:]`.::
+Then we will set up the source
+::
+  srccoord = np.array(self.data.source_coords, dtype=self.dtype)[np.newaxis, :]
+  self.src = SourceLike(name="src", npoint=1, nt=data.traces.shape[1],
+                        dt=self.dt, h=self.model.get_spacing(),
+                        coordinates=srccoord, ndim=len(self.damp.shape),
+                        dtype=self.dtype, nbpml=nbpml)
+  self.src.data[:] = data.get_source()[:, np.newaxis]
+
+SourceLike is an object inheriting PointData, a devito data object for sparse point data
+as a Function symbol.
+
+We initialize the source to be of coordinates : :obj:`srccoord` and set its data to be :obj:`data.get_source()`.
+Receivers are initialized by similar way
+::
+  rec = SourceLike(name="rec", npoint=nrec, nt=nt, dt=dt, h=model.get_spacing(),
+                       coordinates=data.receiver_coords, ndim=len(damp.shape),
+                       dtype=damp.dtype, nbpml=model.nbpml)
 
+Then, We will set up the initial condition and allocate the grid for m.
+Value of m on each grid point is stored as a numpy array in :obj:`m.data[:]`.
+::
   m = DenseData(name="m", shape=model.get_shape_comp(), dtype=damp.dtype)
   m.data[:] = model.padm()
 
-after that, we will initilisize u::
-
+after that, we will initialize u
+::
   u = TimeData(name="u", shape=model.get_shape_comp(), time_dim=nt,
                    time_order=time_order, space_order=spc_order, save=True,
                    dtype=damp.dtype)
 
 TimeData is a devito data object used to store and manage time-varying data
 
```

Comment:
According to the API documentation, it only mention time-varying. Should the doc be changed as well?